services/user.go

```diff
@@ -192,18 +205,93 @@ func (us *User) StartOtpEnroll(userId uint) (secret string, imgStr string, err e
 	return
 }
 
-func (us *User) ValidateOtpEnroll(userId uint, token string) error {
+func (us *User) gerateOtpRecoveryCode(length int) (string, error) {
+	const chars = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+	code := make([]byte, length)
+	for i := range length {
+		num, err := rand.Int(rand.Reader, big.NewInt(int64(len(chars))))
```

Comment:
I think there's actually a crypto.Rand or something that's "better" for this.

Reply:
See the imports, that is `crypto/rand` being used there